contracts/core/modules/RebalancingSetIssuanceModule.sol

```diff
@@ -0,0 +1,455 @@
+     * @param _weth                The address of Wrapped Ether
+     */
+    constructor(
+        address _core,
+        address _vault,
+        address _transferProxy,
+        IWETH _weth
+    )
+        public
+        ModuleCoreState(
+            _core,
+            _vault
+        )
+    {
+        // Commit the WETH instance
+        weth = _weth;
+
+        // Commit the transferProxy instance
+        transferProxy = _transferProxy;
+    }
+
+    /**
+     * Fallback function. Disallows ether to be sent to this contract without data except when
+     * unwrapping WETH.
+     */
+    function ()
+        external
+        payable
+    {
+        require( // coverage-disable-line
+            msg.sender == address(weth),
+            "RebalancingSetIssuanceModule.fallback: Cannot receive ETH directly unless unwrapping WETH"
+        );
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Issue a Rebalancing Set using the base components of the Base Set.
+     * The Base Set is then issued and reissued into the Rebalancing Set. The Base Set quantity issued is calculated
+     * by taking the Rebalancing Set's quantity, unit shares, and natural unit. If the calculated quantity is not
+     * a multiple of the natural unit of the base Set, the quantity is rounded up to the base Set natural unit.
+     *
+     * @param  _rebalancingSetAddress    Address of the rebalancing Set to issue
+     * @param  _rebalancingSetQuantity   The issuance quantity of Rebalancing Set
+     * @param  _keepChangeInVault        Boolean signifying whether excess base Set is transfered to the user 
+     *                                     or left in the vault
+     */
+    function issueRebalancingSet(
+        address _rebalancingSetAddress,
+        uint256 _rebalancingSetQuantity,
+        bool _keepChangeInVault
+    )
+        external
+        nonReentrant
+    {
+        address baseSetAddress = IRebalancingSetToken(_rebalancingSetAddress).currentSet();
+
+        // Calculate required base Set quantity to issue Rebalancing Set
+        uint256 requiredBaseSetQuantity = getBaseSetIssuanceRequiredQuantity(
+            _rebalancingSetAddress,
+            _rebalancingSetQuantity
+        );
+
+        // Issue base Set to this contract, held in this contract
+        coreInstance.issueModule(
+            msg.sender,
+            address(this),
+            baseSetAddress,
+            requiredBaseSetQuantity
+        );
+
+        // Ensure base Set allowance to the transferProxy
+        ERC20Wrapper.ensureAllowance(
+            baseSetAddress,
+            address(this),
+            transferProxy,
+            requiredBaseSetQuantity
+        );
+
+        // Issue rebalancing Set to the sender
+        coreInstance.issueTo(
+            msg.sender,
+            _rebalancingSetAddress,
+            _rebalancingSetQuantity
+        );
+
+        // Return any excess base Set token to the sender
```

Comment:
maybe add comment about either in vault or in wallet depending on boolean indicator